app/src/main/java/com/example/lapnguyen/weather_3/data/source/MainRepositoryImplement.java

```diff
@@ -0,0 +1,31 @@
+package com.example.lapnguyen.weather_3.data.source;
+
+/**
+ * Created by lapnguyen on 16/06/2017.
+ */
+import com.example.lapnguyen.weather_3.api.WeatherService;
+
+import com.example.lapnguyen.weather_3.data.model.Weather;
+import io.reactivex.Observable;
+
+public class MainRepositoryImplement implements MainRepository {
+    WeatherService mWeatherService;
+    private static MainRepository instance;
+
+    public MainRepositoryImplement( WeatherService weatherService) {
+        mWeatherService = weatherService;
+    }
+
+    public static MainRepository getInstance(WeatherService weatherService) {
+        if (instance == null)
+            instance = new MainRepositoryImplement(weatherService);
+        return instance;
+    }
+
+    @Override
+    public Observable<Weather> getWeather(double latitude,double longitude ) {
+
+         return mWeatherService.getWeather(latitude,longitude);
```

Comment:
ERROR: 'method def' child have incorrect indentation level 9, expected level should be 8.